src/renderer/components/hotbar/hotbar-entity-icon.tsx

```diff
@@ -0,0 +1,115 @@
+import "./hotbar-icon.scss";
+
+import React, { DOMAttributes } from "react";
+import { observable } from "mobx";
+import { observer } from "mobx-react";
+import randomColor from "randomcolor";
+
+import { CatalogEntity, CatalogEntityContextMenu, CatalogEntityContextMenuContext } from "../../../common/catalog";
+import { catalogCategoryRegistry } from "../../api/catalog-category-registry";
+import { catalogEntityRegistry } from "../../api/catalog-entity-registry";
+import { navigate } from "../../navigation";
+import { cssNames, IClassName } from "../../utils";
+import { ConfirmDialog } from "../confirm-dialog";
+import { Icon } from "../icon";
+import { HotbarIcon } from "./hotbar-icon";
+
+interface Props extends DOMAttributes<HTMLElement> {
+  entity: CatalogEntity;
+  className?: IClassName;
+  errorClass?: IClassName;
+  remove: (uid: string) => void;
+}
+
+@observer
+export class HotbarEntityIcon extends React.Component<Props> {
```

Comment:
I've separated `HotbarEntityIcon` out of `HotbarIcon` to create another layer of abstraction and make `HotbarIcon` more generic regardless of entity availability.